server/SaveLoad.cpp

```diff
@@ -498,29 +497,29 @@ void LoadPlayerSaveHeaderData(const std::string& filename, std::vector<PlayerSav
         if (!ifs)
             throw std::runtime_error(UNABLE_TO_OPEN_FILE);
 
-        try {
-            // first attempt binary deserialziation
+        std::string signature(5, '\0');
+        if (!ifs.read(&signature[0], 5))
+            throw std::runtime_error(UNABLE_TO_OPEN_FILE);
+        boost::iostreams::seek(ifs, 0, std::ios_base::beg);
+
+        if (strncmp(signature.c_str(), "<?xml", 5)) {
             DebugLogger() << "Attempting binary deserialization...";
             freeorion_bin_iarchive ia(ifs);
 
             ia >> BOOST_SERIALIZATION_NVP(ignored_save_preview_data);
             ia >> BOOST_SERIALIZATION_NVP(ignored_galaxy_setup_data);
             ia >> BOOST_SERIALIZATION_NVP(ignored_server_save_game_data);
             ia >> BOOST_SERIALIZATION_NVP(player_save_header_data);
-
-        } catch (...) {
-            // if binary deserialization failed, try more-portable XML deserialization
-            DebugLogger() << "Trying again with XML deserialization...";
-
-            // reset to start of stream (attempted binary serialization will have consumed some input...)
-            boost::iostreams::seek(ifs, 0, std::ios_base::beg);
+        } else {
+            DebugLogger() << "Attempting XML deserialization...";
```

Comment:
why a log message here but not other similar places?

Reply:
Made log messages symmetrical.